packages/monitor-v2/src/monitor-og/SnapshotVerification.ts

```diff
@@ -217,15 +226,18 @@ const getGraphqlData = async (ipfsHash: string, url: string, retryOptions: Retry
 // We don't want to throw an error if the IPFS request fails for any reason, so we return an Error object instead that
 // will be logged by the bot.
 const getIpfsData = async (ipfsHash: string, url: string, retryOptions: RetryOptions): Promise<unknown | Error> => {
+  let response: Response;
+
+  // Separate try/catch block to catch errors from the fetch() call itself.
   try {
-    const response = await retry(async () => {
-      const fetchResponse = await fetch(`${url}/${ipfsHash}`);
-      if (!fetchResponse.ok) {
-        throw new Error(`Request on ${fetchResponse.url} failed with status ${fetchResponse.status}`);
-      }
-      return fetchResponse;
-    }, retryOptions);
+    response = await retry(async () => await fetch(`${url}/${ipfsHash}`), retryOptions);
+  } catch (error) {
+    assert(error instanceof Error, "Unexpected Error type!");
+    return new IpfsFetchError(error);
+  }
+  if (!response.ok) return new Error(`Request on ${response.url} failed with status ${response.status}`);
```

Comment:
Do you think this `Error` could be an `IpfsFetchError` for better clarity and consistency?

Reply:
IpfsFetchError is used to detect network errors. If the fetch succeeded in establishing connection, but IPFS gateway returned non-ok response code, in most of cases this is due to non existing IPFS CID that we want to dispute.
Unfortunatly, each IPFS gateway provider might behave differently, so we cannot be more granular than this.